common/models/user.js

```diff
@@ -672,13 +672,18 @@ module.exports = function(User) {
       return process.nextTick(cb);
 
     var AccessToken = accessTokenRelation.modelTo;
-
     var query = {userId: {inq: userIds}};
     var tokenPK = AccessToken.definition.idName() || 'id';
     if (options.accessToken && tokenPK in options.accessToken) {
       query[tokenPK] = {neq: options.accessToken[tokenPK]};
     }
-
+    // add principalType in AccessToken.query if using polymorphic relations
+    // between AccessToken and User
+    var isTokenRelationToUserPolymorphic = AccessToken.relations.user.polymorphic &&
+    !AccessToken.relations.user.modelTo;
+    if (isTokenRelationToUserPolymorphic) {
+      query.principalType = this.modelName;
+    }
     AccessToken.deleteAll(query, options, cb);
```

Comment:
we need to make sure we don't simply delete all the accessTokens for given userId since the same userId can exist twice in 2 different user models
for this we add the `principalType` discriminator if we detect the relation between user and accessToken is polymorphic